### PR TITLE
General: Legacy io not used in global plugins

### PR DESCRIPTION
**Pull request by iLLiCiTiT**
## Brief description
Session data from `legacy_io` are not used in most of global plugins.

## Description
Usage of `legacy_io.Session` was replaced with using data stored on pyblish context. Some plugins are still using legacy io but those plugins already have opened PR changing them or replacement in them would need more overall changes.

## Additional information
Replacement usage of session keys stored in context.data:
- `"AVALON_APP"` -> `"hostName"`
- `"AVALON_PROJECT"` -> `"projectName"`
- `"AVALON_ASSET"` -> `"asset"`
- `"AVALON_TASK"` -> `"task"`

## Testing notes:
- [ ] Farm cleanup plugin is working
- [ ] Collect anatomy instance data has right project locally and on farm
- [ ] Collect scene loaded versions  has right project locally and on farm
- [ ] Integrate is integrating to right project locally and on farm
- [ ] Validate editorial asset is validating correctly

### CODE REVIEW

**Review by kalisp — CHANGES_REQUESTED** [review]
In AE it fails locally with 
2022-12-19 13_52_25-Testing Platform PC - AnyDesk

(It probably fails also in PS, Maya and Nuke - in automatic tests)

File: openpype/plugins/publish/integrate.py [hunk 1]
```diff
@@ -244,7 +243,7 @@ def filter_representations(self, instance):
         return filtered_repres
 
     def register(self, instance, file_transactions, filtered_repres):
-        project_name = legacy_io.active_project()
+        project_name = instance.context["projectName"]
```
Comment (iLLiCiTiT) [inline openpype/plugins/publish/integrate.py:246]:
```suggestion
        project_name = instance.context.data["projectName"]
```

**Comment by iLLiCiTiT** [comment]
> In AE it fails locally with

Should be fixed now.

**Review by jakubjezek001 — APPROVED** [review]
Tested only Validate Editorial asset name and it worked

**Review by antirotor — APPROVED** [review]
tested in maya and it works

**Pull request merged**